maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,261 @@ title: Monitor Calico component metrics
+Your prometheus configuration is going to check services and catch endpoints related to felix-demo-svc service, you need to create a service with that name to expose Felix metrics. This will help in discovery of your services and **makes** it dynamic. 
 
-#### Best practices
+> **Note**: Felix uses port 9091 TCP to publish its metrics.
+   {: .alert .alert-info}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: felix-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-node
+  ports:
+  - port: 9091
+    targetPort: 9091
+EOF
+```
+
+## 4. Creating a service to expose Typha metrics
+
+You need to create a service that exposes your typha instance metrics to Prometheus.
+
+> **Note**: Typha uses **port 9091** TCP to publish its metrics. However, in AWS Implementation of calico port is set to TCP Port 9093 via **TYPHA_PROMETHEUSMETRICSPORT** environment variable.
+   {: .alert .alert-warning}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: typha-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-typha
+  ports:
+  - port: 9093
+    targetPort: 9093
+EOF
+```
+
+## 5. Creating Prometheus pod
+
+Now that you allowed the user to view the metrics and have a valid config file for your Prometheus, it's time to create the pod.
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prometheus-pod
+  labels:
+    app: prometheus-pod
+    role: monitoring
+spec:
+  containers:
+  - name: prometheus-pod
+    image: prom/prometheus
+    resources:
+      limits:
+        memory: "128Mi"
+        cpu: "500m"
+    volumeMounts:
+    - name: config-volume
+      mountPath: /etc/prometheus/prometheus.yml
+      subPath: prometheus.yml
+    ports:
+    - containerPort: 9090
+  volumes:
+  - name: config-volume
+    configMap:
+      name: prometheus-config
+EOF
+```
+
+Check your cluster running pods to assure pod creation was successful.
+    
+    
+```bash
+kubectl get pods prometheus-pod
+```
+It should return something like the following.
+
+```
+NAME             READY   STATUS    RESTARTS   AGE
+prometheus-pod   1/1     Running   0          16s
+```
+
+You can access prometheus dashboard by using port-forwarding feature.
+
+```bash
+kubectl port-forward pod/prometheus-pod 9090:9090
+```
+
+Browse to [http://localhost:9090](http://localhost:9090) you should be able to see prometheus dashboard.
+
+## 6. Enabling Felix metrics
+
+Prometheus metric microservice is set to **false** by default. You have to manually change your Felix configuration (**prometheusMetricsEnabled**) via calicoctl in order to use this feature.
+
+> **Note**: A comprehensive list of configuration values can be [found at this link]({{ site.baseurl }}/reference/felix/configuration).
+   {: .alert .alert-info}
+
+
+
+```bash
+calicoctl replace -f - <<EOF
```

Comment:
It might be better to use `calicoctl patch` here instead.  Otherwise you are overwriting any other config they may have set on their cluster, which could break things for them.